Sources/Sentry/SentryTimeToDisplayTracker.m

```diff
@@ -140,6 +142,9 @@ - (void)framesTrackerHasNewFrame:(NSDate *)newFrameDate
         if (!_waitForFullDisplay) {
             [SentryDependencyContainer.sharedInstance.framesTracker removeListener:self];
 #    if SENTRY_TARGET_PROFILING_SUPPORTED
+            if (SentrySDK.options.enableContinuousProfiling) {
```

Comment:
`m`: We already pass in `enableWaitForFullDisplay` via the constructor, which stems from the options. It would be great to follow this pattern.

Reply:
why not go the other way and pass in the options instance?

Reply:
Here it is in case that's ok with you: https://github.com/getsentry/sentry-cocoa/pull/3989

Reply:
So, that PR has some failing tests, and I'm guessing they're due to the mismatch I mentioned. I think that's the right way to make this change, but I don't currently have the time to focus on that. I unstuck this branch by making the new boolean init param and passing it from `SentryUIViewControllerPerformanceTracker`, which was already grabbing the options from SentrySDK.options in its own init method. That would go away in the other PR. Please have a look at the failing tests there and see if there are some quick fixes you could help with!

Reply:
I'm confused why you went with the refactoring in https://github.com/getsentry/sentry-cocoa/pull/3989, while just passing in the argument similar to `enableWaitForFullDisplay` would be way less work. I would ditch https://github.com/getsentry/sentry-cocoa/pull/3989 right now. You can also keep the access to `SentrySDK.options.enableContinuousProfiling` in this PR, if you're in a hurry. Most important would be some tests.

Reply:
Ah I see, because `SentryUIViewControllerPerformanceTracker` is a singleton, and we would need to refactor it. The `SentryTimeToDisplayTracker` isn't a singleton and requires its dependencies via the constructor, so it's easier to test. Nevermind.